task_processing/interfaces/event.py

```diff
@@ -1,13 +1,18 @@
 from pyrsistent import field
+from pyrsistent import m
+from pyrsistent import PMap
 from pyrsistent import PRecord
 
-# TODO: organize and explain these
+from task_processing.interfaces.task_executor import TaskConfig
 
 
 class Event(PRecord):
     # reference to platform-specific event object
     raw = field()
     # is this the last event for a task?
     terminal = field(type=bool, mandatory=True)
+    success = field(type=(bool, type(None)), initial=None)
     # platform-specific event name
     platform_type = field(type=str)
+    task_config = field(type=TaskConfig)
+    middleware_data = field(type=PMap, initial=m())
```

Comment:
im also not really convinced by this. I kind of get that you need to store data somewhere, but the event seems the wrong place to do it.